pkg/apis/core/v1beta1/types.go

```diff
@@ -351,6 +371,130 @@ type StorageProfileList struct {
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataSourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items provides a list of DataSources
+	Items []DataSource `json:"items"`
+}
+
+// DataImportCron defines a cron job for recurring polling/importing disk images as PVCs into a golden image namespace
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type DataImportCron struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataImportCronSpec   `json:"spec"`
+	Status DataImportCronStatus `json:"status,omitempty"`
+}
+
+// DataImportCronSpec defines specification for DataImportCron
+type DataImportCronSpec struct {
+	// Source specifies where to poll disk images from
+	Source DataImportCronSource `json:"source"`
+	// Schedule specifies in cron format when and how often to look for new imports
+	Schedule string `json:"schedule"`
+	// GarbageCollect specifies whether old PVCs should be cleaned up after a new PVC is imported
+	// Options are currently "Never" and "Outdated", defaults to "Never"
+	// +optional
+	GarbageCollect *DataImportCronGarbageCollect `json:"garbageCollect,omitempty"`
+	// ManagedDataSource specifies the name of the corresponding DataSource this cron will manage
+	ManagedDataSource string `json:"managedDataSource"`
+}
+
+// DataImportCronGarbageCollect represents the DataImportCron garbage collection mode
+type DataImportCronGarbageCollect string
+
+const (
+	// DataImportCronGarbageCollectNever specifies that garbage collection is disabled
+	DataImportCronGarbageCollectNever DataImportCronGarbageCollect = "Never"
+	// DataImportCronGarbageCollectOutdated specifies that old PVCs should be cleaned up after a new PVC is imported
+	DataImportCronGarbageCollectOutdated DataImportCronGarbageCollect = "Outdated"
+)
+
+// DataImportCronSource defines where to poll and import disk images from
+type DataImportCronSource struct {
+	Registry *DataVolumeSourceRegistry `json:"registry"`
+}
+
+// DataImportCronStatus provides the most recently observed status of the DataImportCron
+type DataImportCronStatus struct {
+	// LastImportedPVC is the last imported PVC
+	LastImportedPVC *DataVolumeSourcePVC `json:"lastImportedPVC,omitempty"`
+	// LastExecutionTimestamp is the time of the last polling
+	LastExecutionTimestamp *metav1.Time `json:"lastExecutionTimestamp,omitempty"`
+	// LastImportTimestamp is the time of the last import
+	LastImportTimestamp *metav1.Time              `json:"lastImportTimestamp,omitempty"`
+	Conditions          []DataImportCronCondition `json:"conditions,omitempty" optional:"true"`
+}
+
+// DataImportCronCondition represents the state of a data import cron condition
+// FIXME: needs LastTransitionTime/LastHeartbeatTime/Reason?
```

Comment:
What was the reason of not adding them right now?

Reply:
Yeah, may as well add them now

Reply:
Added